### ISSUE TITLE
gtk linking problem with lasted git

### ISSUE THREAD

**Issue by mckaygerhard** [Yabause/yabause#70]
i now compiling the lasted git but when do have a strange problem:

using gcc 4.2, gtk 2.18, no qt, mesa 7.0, cmake 2.8.0

the lasted error seems easy to fix, due are a deprecated problem in my gtk libs, but previos problems i cannot solve by myselft, please provide me more info or solve it to me

```
[100%] Building C object src/gtk/CMakeFiles/yabause-gtk.dir/yuiwindow.c.o
Linking C executable yabause
../libyabause.a(cs2.c.o): In function `Cs2ReadLong':
cs2.c:(.text+0x4be6): undefined reference to `__builtin_bswap32'
../libyabause.a(vdp1.c.o): In function `Vdp1RamWriteLong':
vdp1.c:(.text+0xa6b): undefined reference to `__builtin_bswap32'
../libyabause.a(vdp1.c.o): In function `Vdp1FrameBufferReadLong':
vdp1.c:(.text+0xabf): undefined reference to `__builtin_bswap32'
../libyabause.a(vdp1.c.o): In function `Vdp1FrameBufferWriteLong':
vdp1.c:(.text+0xf4a): undefined reference to `__builtin_bswap32'
../libyabause.a(vdp1.c.o): In function `Vdp1RamReadLong':
vdp1.c:(.text+0xf78): undefined reference to `__builtin_bswap32'
../libyabause.a(vdp2.c.o):vdp2.c:(.text+0xf3a): more undefined references to `__builtin_bswap32' follow
/usr/lib/gcc/i486-linux-gnu/4.2.0/../../../../lib/libgtkglext-x11-1.0.so: undefined reference to `gtk_widget_get_realized'
collect2: ld devolvió el estado de salida 1
make[2]: *** [src/gtk/yabause] Error 1
make[1]: *** [src/gtk/CMakeFiles/yabause-gtk.dir/all] Error 2
make: *** [all] Error 2
```

i'm compiling for xanderos eepc children flavor

in any case, the symbol `gtk_widget_get_realized' seems valid on my gtk gl extension lib...

any info about that ?

the 0.9.13 version compiles for me with gcc 4.1 and 4.2 with more older gtk libs

**Comment by Guillaumito** [comment]
looks like your gtk and gtkglext versions mismatch or something as gtk_widget_get_realized was added in gtk 2.20 and so, doesn't exists in 2.18: https://developer.gnome.org/gtk2/stable/GtkWidget.html#gtk-widget-get-realized

**Comment by mckaygerhard** [comment]
arrrggg i cannot compile in xandros eepc and embebed own device, umm 

theres another way to verify or determines whether widget is realized/contructed in the memory?

i cannot find in gtk api a way todo this, may can i help and i can made a patch and  also some test (i have a very large amount of games to test)

**Comment by mckaygerhard** [comment]
i find a solution, due its a posibility to made compille-able to older gtk vesion :

GTK_WIDGET_REALIZED() are equivalen so i will made a patch to able to build if a gtk older are detected.

**Comment by Guillaumito** [comment]
The bswpa32 problem is now fixed.

**Comment by mckaygerhard** [comment]
fixed in commit 07076d810287a84f55cd0884c9ee93bc4eb3e05f
